interfaces/builtin/content.go

```diff
@@ -78,6 +78,16 @@ func (iface *contentInterface) BeforePrepareSlot(slot *snap.SlotInfo) error {
 		slot.Attrs["content"] = slot.Name
 	}
 
+	// Error if "read" or "write" are present alongside "source".
+	if _, ok := slot.Attrs["source"]; ok {
```

Comment:
You can use `Attr(name, value) error` on PlugInfo/SlotInfo now, would be better for consistency, and also we might want o hide Attrs map in near future.

Reply:
This is not plug/slot info but an attribute loaded from Attrs already

Reply:
No, this **is** SlotInfo (and PlugInfo, respectively, in BeforePreparePlug below), so you can use Attrer interface over it via `if err := slot.Attr("source", &source); err {...`. Note that `path(...)` helper does exactly that and works both for Plug/SlotInfo and ConnectedPlug/Slot.

Reply:
Ah sorry, I misread the diff, you are totally right

Reply:
Done